crates/uv-workspace/src/pyproject.rs

```diff
@@ -113,6 +113,20 @@ impl AsRef<[u8]> for PyProjectToml {
     }
 }
 
+/// A specifier item in a PEP 735 Dependency Group
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+#[serde(rename_all = "kebab-case", untagged)]
+pub enum DependencyGroupSpecifier {
+    /// PEP 508 requirement string
+    Requirement(String),
+    /// Include another dependency group
+    #[serde(rename_all = "kebab-case")]
+    Table {
+        /// The name of the group to include
+        include_group: Option<String>,
+    },
```

Comment:
Do we... want a catch-all variant? So we can catch other objects that are added in the future...? (We might want to error in that case anyway.)

Reply:
I'm not sure. Sounds like a nice property sometimes? We don't do this elsewhere in the `pyproject.toml` do we?

Reply:
Ok the current structure is pretty strict... It rejects empty tables, extra fields, and missing `include_group`... It does match the reference implementation: https://peps.python.org/pep-0735/#reference-implementation. But I'd say what I have here is too strict though since we're doing eager validation. I think we should just warn if we see unsupported groups.